prov/rxd/src/rxd_cq.c

```diff
@@ -905,14 +905,18 @@ static void rxd_progress_buf_pkts(struct rxd_ep *ep, fi_addr_t peer)
 	size_t msg_size;
 	struct rxd_x_entry *rx_entry = NULL;
 	struct rxd_data_pkt *data_pkt;
+	struct rxd_peer *peer_ptr, *peer_ptr1;
 
-	while (!dlist_empty(&ep->peers[peer].buf_pkts)) {
-		pkt_entry = container_of((&ep->peers[peer].buf_pkts)->next,
+
+	peer_ptr = rxd_ep_getpeer(ep, peer);
+
+	while (!dlist_empty(&(peer_ptr->buf_pkts))) {
+		pkt_entry = container_of((&(peer_ptr->buf_pkts))->next,
 					struct rxd_pkt_entry, d_entry);
 		base_hdr = rxd_get_base_hdr(pkt_entry);
-		if (base_hdr->seq_no != ep->peers[peer].rx_seq_no)
+		if (base_hdr->seq_no != peer_ptr->rx_seq_no)
```

Comment:
This change works, but it may be easier to do:
```
if (base_hdr->seq_no != rxd_getpeer(ep, peer)->rx_seq_no)
```
That way you don't need to declare extra local variables and do assignments in other places.

Reply:
👍 thanks

Reply:
added to other files.